### PR TITLE
finished homework 16 (tomcat, hibernate, servlets, html, css)

### PR DESCRIPTION
**Pull request by Sofasmile**

### CODE REVIEW

File: src/main/java/com/tomcat/dao/StudentDao.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.tomcat.dao;
+
+import com.tomcat.entity.Student;
+
+import java.util.List;
+
+public interface StudentDao {
+    void insertStudent(Student student);
```
Comment (Anderwerty) [inline src/main/java/com/tomcat/dao/StudentDao.java]: You have duplicate word student. You have it already in the name of dao.
insertStudent ->insert (or create)
updateStudent -> update
deleteStudent->deleteById
getByIdStudent-> getById (or findById)
getAllStudents ->getAll (or findAll)

File: src/main/java/com/tomcat/dao/impl/StudentDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        return student;
+    }
+
+    @Override
+    public List<Student> getAllStudents() {
+        Session session = null;
+        List students = new ArrayList<Student>();
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            students = session.createQuery("SELECT a FROM Student a", Student.class).getResultList();
```
Comment (Anderwerty) [inline src/main/java/com/tomcat/dao/impl/StudentDaoImpl.java]: Could we use a constant for SELECT a FROM Student a?

File: src/main/java/com/tomcat/dao/impl/StudentDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        } catch (Exception e) {
+            log.error("Get all error: " + e.getMessage());
+        } finally {
+            session.close();
+        }
+        return students;
+    }
+
+    public static synchronized StudentDao getInstance() {
```
Comment (Anderwerty) [inline src/main/java/com/tomcat/dao/impl/StudentDaoImpl.java]: Maybe better to use a double check?

**Review by Anderwerty — CHANGES_REQUESTED** [review]

File: src/main/java/com/tomcat/controller/CreateController.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        String name = request.getParameter("name");
+        Integer age = Integer.valueOf(request.getParameter("age"));
+        Course course = Course.valueOf(request.getParameter("course"));
+        String speciality = request.getParameter("speciality");
+        Student student = new Student()
+                .setName(name)
+                .setAge(age)
+                .setCourse(course)
+                .setSpeciality(speciality);
+        if (Objects.equals(studentId, null))
```
Comment (Anderwerty) [inline src/main/java/com/tomcat/controller/CreateController.java]: Why not just   studentId ==null or Objects.isNull

**Review by Anderwerty — CHANGES_REQUESTED** [review]

File: src/main/java/com/tomcat/controller/CreateController.java (hunk 4 above)
Comment (Sofasmile) [reply src/main/java/com/tomcat/controller/CreateController.java]: I thinked it look like better. May I use this construction: Objects.isNull(studentId)? Alexandr said us about it last lesson.

**Review by Anderwerty — APPROVED** [review]

**Pull request merged**